broker/fragment/store_azure.go

```diff
@@ -29,30 +35,63 @@ type azureBackend struct {
 	endpoint    string
 	accountName string
 	client      pipeline.Pipeline
-	credentials *azblob.SharedKeyCredential
+	svcClient   service.Client
 	clientMu    sync.Mutex
+	udc         *service.UserDelegationCredential
+	udc_exp     *time.Time
 }
 
 func (a *azureBackend) Provider() string {
 	return "azure"
 }
 
+// Cache UserDelegationCredentials and refresh them when needed
+func (a *azureBackend) GetUserDelegationCredential() (*service.UserDelegationCredential, error) {
```

Comment:
this should be lower-case so it's not exported. Also, it should live near the bottom as utility / implementation detail. Important code first.

Also, does this need to have an expiration time?  Is that enforced by Azure ? Are these minting unique credential keys? Could we mint one that has the process's lifetime? We never write these out or transmit them anywhere so I'm unclear what the short expiration is seeking to protect but perhaps Azure requires it?

Reply:
It needs an expiry. This is creating a signing key that lets you statelessly sign specific blob requests, so I assume they want expiration for the same reason e.g a JWT needs expiration: the fallout of a time-limited key getting leaked is less than an unlimited key.

[The docs](https://learn.microsoft.com/en-us/rest/api/storageservices/get-user-delegation-key#request-body) say:

Element | Description
-- | --
Start | Required. The start time for the user delegation SAS, in ISO date format. It must be a valid date and time within seven days of the current date.
Expiry | Required. The expiration time of the user delegation SAS, in ISO date format. It must be a valid date and time within seven days of the current date.

Reply:
Changed to lower case and moved near the bottom